runtime/lua/vim/lsp/handlers.lua

```diff
@@ -215,9 +215,14 @@ M['textDocument/documentSymbol'] = symbol_handler
 M['workspace/symbol'] = symbol_handler
 
 --@see https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_rename
-M['textDocument/rename'] = function(_, _, result)
-  if not result then return end
-  util.apply_workspace_edit(result)
+M['textDocument/rename'] = function(_, _, result, client_id)
+  if not result or not util.workspace_edit_has_edits(result) then
```

Comment:
```suggestion
  if not util.workspace_edit_has_edits(result) then
```

`not result` is redundant here.

Reply:
Fixed.